data/tools/python/src/inaira/data/frame_producer.py

```diff
@@ -0,0 +1,261 @@
+            self.config.ready_endpoint
+        )
+
+        # Create the frame release channel, bind it to the endpoint and set the default
+        # subscription on the channel
+        self.release_channel = IpcChannel(IpcChannel.CHANNEL_TYPE_SUB)
+        self.release_channel.bind(self.config.release_endpoint)
+        self.release_channel.subscribe(b'')
+        self.logger.debug(
+            "Created frame release IPC channel on endpoint %s",
+            self.config.release_endpoint
+        )
+
+        # Create a thread to handle frame release messages
+        self.release_thread = threading.Thread(target=self.process_release)
+
+        # Create the queue to contain available buffer IDs and precharge it with all available
+        # shared memory buffers
+        self.free_buffer_queue = queue.Queue()
+        for idx in range(self.shared_buffer_manager.get_num_buffers()):
+            self.free_buffer_queue.put(idx)
+        self.logger.debug(
+            "Created free buffer queue with %d buffers pre-charged",
+            self.free_buffer_queue.qsize()
+        )
+
+        # Initialise msg index counter
+        self._next_msg_id = 0
+
+        # Set the internal run flag to true
+        self._run = True
+
+    def get_next_msg_id(self):
+        """Increment and return the next message ID for IPC messages."""
+        self._next_msg_id += 1
+        return self._next_msg_id
+
+    def get_dtype_enumeration(random, img_dtype):
+        list_of_dtypes = ["unknown","uint8","uint16","uint32","uint64","float"]
+        # Is theee data type contained within the odin data eenuumeeration list?
+        if img_dtype in list_of_dtypes:
+            enumerated_dtype = list_of_dtypes.index(img_dtype)
+
+        else:
+            enumerated_dtype = 0
+        
+        # Return the enumerated data type
+        return enumerated_dtype
+
+    def run(self):
+        """Run the frame producer main event loop."""
+
+        # Start the release channel processing thread
+        self.release_thread.start()
+
+        # Allow the IPC channels to connect and then notify the frame processor of the buffer
+        # configuration
+        time.sleep(1.0)
+        self.notify_buffer_config()
+
+        # Set the number of bytes for a simulated frame to the shared buffer size. This will need
+        # to be modified once a frame header is defined and used
+        num_bytes = self.config.shared_buffer_size
+
+        #Load the test images
+        testfilespath = self.config.testfile_path 
+        self.logger.debug("testfilespath = " + testfilespath)
+
+        totalimages = 0
+
+        # Number of files in directory
+        for files in walk(testfilespath):
+            for Files in files:
+                totalimages += 1
+
+        # Loop over the specified number of frames and transmit them
+        self.logger.info("Sending %d frames to processor\n", self.config.frames)
+        for frame in range(self.config.frames):
```

Comment:
One thing I think would be really useful is having an option to set the number of frames in the config file to 0. In this case, you could have the producer run indefinitely, until interrupted by the user.